File: java/src/tarehart/rlbot/steps/rotation/RotationUtil.java
package tarehart.rlbot.steps.rotation;

import tarehart.rlbot.math.vector.Vector3;

public class RotationUtil {

    public static double maxOrbitHeightAbovePlane(Vector3 axisOfRotation, Vector3 planeNormal) {
        double xVal = axisOfRotation.projectToPlane(planeNormal).magnitude();
        double angleAbovePlane = Math.acos(xVal);
        return Math.cos(angleAbovePlane);
    }

    public static double shortWay(double radians) {

        radians = radians % (Math.PI * 2);
        if (radians > Math.PI) {
            radians -= Math.PI * 2;
        }
        return radians;
    }

}

File: src/main/java/tarehart/rlbot/steps/rotation/RotationUtil.kt
package tarehart.rlbot.steps.rotation

import tarehart.rlbot.math.vector.Vector3

object RotationUtil {

    fun maxOrbitHeightAbovePlane(axisOfRotation: Vector3, planeNormal: Vector3): Double {
        val xVal = axisOfRotation.projectToPlane(planeNormal).magnitude()
        val angleAbovePlane = Math.acos(xVal)
        return Math.cos(angleAbovePlane)
    }

    fun shortWay(initialRadians: Double): Double {
        var radians = initialRadians

        radians %= (Math.PI * 2)
        if (radians > Math.PI) {
            radians -= Math.PI * 2
        }
        return radians
    }

}
